### ISSUE TITLE
fs::metadata(entry_path).file_type().is_symlink() not work

### ISSUE THREAD

**Issue by xiaozhuai** [rust-lang/rust#39088]
```rust
let metadata = fs::metadata(file_path).try(stderr);
let filetype = metadata.file_type();
println!("{}", filetype.is_symlink());
```

It's always false.

**Comment by xiaozhuai** [comment]
I'm on Linux Mint, the kernel version is `4.4.0-21-generic` `#37-Ubuntu SMP Mon Apr 18 18:33:37 UTC 2016`

**Comment by xiaozhuai** [comment]
The file system is ext4

**Comment by nagisa** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/29302; not a bug.

**Comment by brson** [comment]
Update the docs for `is_symlink` to note that a common source of confusion is to call this on the result of `fs::metadata` not `fs::symlink_metadata`.

**Comment by CartesianDaemon** [comment]
Hi. I'm new to rust, and have been looking through the E-easy tagged bugs for one it would be easy to start with. (Thank you very much for making easy bugs available as on-ramps, it is very very much appreciated.)

I think the appropriate change is in src/libstd/fs.rs in the documentation for is_symlink(). Adding a second line to the description warning that you should use symlink_metadata by default, and changing the example to use fs::symlink_metadata() not fs::metadata().

I have made and tested that change locally. Does that sound right?

**Comment by nagisa** [comment]
> Does that sound right?

Yes. I would suggest just posting the PR. Reviewing (commenting on) code directly both spares you the effort describing the changes and us from parsing the description.

**Comment by phungleson** [comment]
I believe this one is done?

**Comment by CartesianDaemon** [comment]
Assuming I'm right, yes, I think my commit fixed this particular issue, did everything suggested, and is now merged safely to master.

**Comment by frewsxcv** [comment]
Addressed by @CartesianDaemon in #39176. Thanks again!
